wallet/src/actors/app/methods.rs

```diff
@@ -607,6 +628,24 @@ impl App {
         });
     }
 
+    /// Offload node status into a worker that operates on a different Arbiter than the main
+    /// server thread, so as not to lock the rest of the application.
+    pub fn handle_change_status_in_worker(
+        &self,
+        status: &types::StateMachine,
+        wallet: &types::SessionWallet,
+        sink: types::DynamicSink,
+    ) {
+        // NOTE: Possible enhancement.
+        // Maybe is a good idea to use a shared reference Arc instead of cloning `block` altogether,
+        // moreover when this method is called iteratively by `handle_block_notification`.
```

Comment:
This method has nothing to do with `handle_block_notification` :wink: 
```suggestion
        // Maybe is a good idea to use a shared reference Arc instead of cloning `block` altogether
```